app/commands/track/update_median_wait_times.rb

```diff
@@ -0,0 +1,30 @@
+class Track
+  class UpdateMedianWaitTimes
+    include Mandate
+
+    def call
+      Track.active.find_each do |track|
+        track.update(median_wait_time: median_wait_time(track))
+      end
+    end
+
+    private
+    def median_wait_time(track)
+      wait_times = Mentor::Discussion.
+        joins(:request).
+        joins(:exercise).
+        where('exercise.track_id': track.id).
+        select('TIMESTAMPDIFF(SECOND, mentor_requests.created_at, mentor_discussions.created_at) AS wait_time').
+        map(&:wait_time)
```

Comment:
This executes the following query:

```sql
SELECT TIMESTAMPDIFF(SECOND, mentor_requests.created_at, mentor_discussions.created_at) AS wait_time
FROM `mentor_discussions`
INNER JOIN `mentor_requests` ON `mentor_requests`.`id` = `mentor_discussions`.`request_id`
INNER JOIN `solutions` ON `solutions`.`id` = `mentor_discussions`.`solution_id`
INNER JOIN `exercises` `exercise` ON `exercise`.`id` = `solutions`.`exercise_id`
WHERE `exercise`.`track_id` = 3
```

This query uses only PRIMARY keys as is thus quite fast.